Realm/Realm/DatabaseTypes/RealmValue.cs

```diff
@@ -805,6 +941,9 @@ public override int GetHashCode()
                     RealmValueType.Decimal128 => AsDecimal128().GetHashCode(),
                     RealmValueType.ObjectId => AsObjectId().GetHashCode(),
                     RealmValueType.Object => AsIRealmObject().GetHashCode(),
+                    RealmValueType.List => AsList().GetHashCode(),
+                    RealmValueType.Set => AsSet().GetHashCode(),
+                    RealmValueType.Dictionary => AsDictionary().GetHashCode(),
```

Comment:
I don't think I saw a test for hashcode/equals, so it'd be worth adding it.

Reply:
We have tests for equals, but not for hashcode. I suppose in the test you wrote in the comment you wanted to compare the original list to the realm value? 
As it is written the hashcode should be the same if the list is unmanaged, but will be different if the list is managed.

Reply:
Yeah, I was interested in the managed case - particularly because we're fetching a new collection instance on each property access, I don't think the hashcode function is ideal here, though I don't have a good suggestion for how to do it better. I guess one could argue it's probably not super important since we've never had stable hashcodes for collections and people haven't complained about it.